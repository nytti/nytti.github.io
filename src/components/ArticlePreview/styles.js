export default {
  action: {
    width: '100%',
  },
  media: {
    height: 220,
  },
  link: {
    textDecoration: 'none',
  }
}
