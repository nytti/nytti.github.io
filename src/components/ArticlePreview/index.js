// One item in news list. This component is stateful and little tricky -
// it handles news item modal dialog rendering to make system architecture
// more encapsulated

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

// Material UI
import { withStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardMedia from '@material-ui/core/CardMedia'
import CardContent from '@material-ui/core/CardContent'
import CardActionArea from '@material-ui/core/CardActionArea'
import Typography from '@material-ui/core/Typography'

import { nyt } from 'config/config'

// Styles
import styles from './styles'

const { imageUrl } = nyt

export default
@withStyles(styles)
class ArticlePreview extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    article: PropTypes.object.isRequired,
    topic: PropTypes.string.isRequired,
  }

  render() {
    const { article, classes, topic } = this.props
    const {
      _id: id,
      multimedia,
      headline: {
        main: title,
      },
    } = article
    const media = multimedia.find(({ subType }) => subType === 'master768') || multimedia[0]

    return (
      <Card className={classes.root}>
        <Link to={`/${id}`} className={classes.link}>
          <CardActionArea className={classes.action}>
            <CardContent>
              <Typography>
                { title }
              </Typography>
            </CardContent>
            { media && (
              <CardMedia
                className={classes.media}
                image={`${imageUrl}${media.url}`}
                title={title}
              />
            )}
            <CardContent>
              <Typography component="p" color="textSecondary">
                { topic }
              </Typography>
            </CardContent>
          </CardActionArea>
        </Link>
      </Card>
    )
  }
}
