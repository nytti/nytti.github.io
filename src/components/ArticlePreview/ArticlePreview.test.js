import React from 'react'
import { shallow } from 'enzyme'

import items from 'articles/items'

import ArticlePreview from './index'

describe('<ArticlePreview />', () => {
  test('<ArticlePreview /> snapshot', () => {
    const element = shallow(<ArticlePreview article={items[0]} topic="test" />)
    expect(element).toMatchSnapshot()
  })
})
