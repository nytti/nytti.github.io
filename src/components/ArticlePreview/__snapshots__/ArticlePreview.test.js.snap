// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticlePreview /> <ArticlePreview /> snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(ArticlePreview)
    article={
      Object {
        "_id": "Item-Id-Test",
        "document_type": "article",
        "headline": Object {
          "main": "Item-Headline-Main-Test",
        },
        "multimedia": Array [
          Object {
            "url": "image-url-test",
          },
        ],
        "pub_date": "2018-09-24T15:58:39+0000",
        "snippet": "Item-Snippet-Test",
        "web_url": "https://test-item.com",
      }
    }
    topic="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "article": Object {
        "_id": "Item-Id-Test",
        "document_type": "article",
        "headline": Object {
          "main": "Item-Headline-Main-Test",
        },
        "multimedia": Array [
          Object {
            "url": "image-url-test",
          },
        ],
        "pub_date": "2018-09-24T15:58:39+0000",
        "snippet": "Item-Snippet-Test",
        "web_url": "https://test-item.com",
      },
      "classes": Object {
        "action": "ArticlePreview-action-1",
        "link": "ArticlePreview-link-3",
        "media": "ArticlePreview-media-2",
      },
      "topic": "test",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "article": Object {
          "_id": "Item-Id-Test",
          "document_type": "article",
          "headline": Object {
            "main": "Item-Headline-Main-Test",
          },
          "multimedia": Array [
            Object {
              "url": "image-url-test",
            },
          ],
          "pub_date": "2018-09-24T15:58:39+0000",
          "snippet": "Item-Snippet-Test",
          "web_url": "https://test-item.com",
        },
        "classes": Object {
          "action": "ArticlePreview-action-1",
          "link": "ArticlePreview-link-3",
          "media": "ArticlePreview-media-2",
        },
        "topic": "test",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
